### ISSUE TITLE
XMA files will not play in foobar2000

### ISSUE THREAD

**Issue by dwalkerdon** [kode54/vgmstream#141]
here are some XMA files that will not work in foobar

**Comment by dwalkerdon** [comment]
for some reason i cant attach any files to this message

**Comment by richardpl** [comment]
Then use numerous site for uploading files.

**Comment by dwalkerdon** [comment]
alright here is a link https://mega.nz/#!QeAw1RSQ!4kFhg4jRKZ5lRZBhc_ySaAbyG7s1GgynxDDJ7ocPlgs

**Comment by richardpl** [comment]
I only can tell you that file decodes just fine with ffmpeg.

**Comment by dwalkerdon** [comment]
why doesnt it work in vgmstream in foobar2000?

**Comment by dwalkerdon** [comment]
i'm going to download ffmpeg and see if it works

**Comment by kode54** [comment]
Your file is stuck "starting" on Mega for me, maybe you could try uploading it here again?

**Comment by dwalkerdon** [comment]
https://mega.nz/#!UDA3iBST!4kFhg4jRKZ5lRZBhc_ySaAbyG7s1GgynxDDJ7ocPlgs

**Comment by kode54** [comment]
To something other than Mega. To this page. By dragging and dropping the file onto a post, as it says.

**Comment by dwalkerdon** [comment]
here is the entire folder https://mega.nz/#F!wCxSwIgZ!ZNf8Ell4Z5c5sJHZjBkaaQ

**Comment by dwalkerdon** [comment]
I am going to upload it using mediafire: https://www.mediafire.com/folder/xkd9bndit6i79/Jeopardy!_(2012-10-16)(-)(THQ) 

https://www.mediafire.com/folder/2lqcm43mqzada/Jeopardy!%20(2012-10-16)(-)(THQ)

**Comment by kode54** [comment]
VGMStream is failing on these files, which must have been converted in the ripping process, as they have RIFF headers which specify a total file size which is larger than the actual files. They'll need to be fixed.

**Comment by dwalkerdon** [comment]
i downloaded ffmpeg, and it does not play the xma files.

**Comment by dwalkerdon** [comment]
this set on x360joshw.info also has trouble: Wheel of Fortune (2012-10-16)(-)(THQ).7z    http://x360.joshw.info/w/Wheel%20of%20Fortune%20(2012-10-16)(-)(THQ).7z

**Comment by bnnm** [comment]
All those files, and many more in joshw, are **wrong**. 

They were modified (transmogrified as I like to call them) with some tool, for toWav I guess.

To get them playing you should fix the RIFF header so it has proper values, probably some stuff exists out there.

I hate that stuff with irrational passion. While it'd be trivial to add support, I hope nobody never, ever does. And if it comes to it (please don't), at least put a giant comment like "THIS IS FOR GARBAGE, INCOMPETENTLY CONVERTED XMA, THAT IN NO WAY EXIST IN REAL GAMES".

**Comment by kode54** [comment]
I will not be fixing this in VGMStream, mainly because it looks like these files are even incomplete, if they're missing at least loop information, or other useful data present in the XMA headers.

**Comment by richardpl** [comment]
FFmpeg does play those files, give actual full console output.

**Comment by bnnm** [comment]
Sure, it's possible to play if you skip the RIFF size validations (surprising FFmpeg would do that though).

I just don't think there is value in supporting a badly formed XMA that can't even get RIFF right (a simple, ancient spec) and no encoder would ever create.

I relax validations here and there for many formats so it's not like I mind data integrity that much, but those XMA are just terrible, plus they always lose loop info which is like the main vgmstream thing?


(BTW @richardpl, I hope you could take a look [at this XMA issue](https://trac.ffmpeg.org/ticket/6722) sometime in the future, as it affects looping a bit)

**Comment by richardpl** [comment]
@bnnm isnt that because skip samples at beginning and end is ignored? Almost same code is in wmalosslessdec.c

**Comment by bnnm** [comment]
@richardpl: not just skip samples, there are 2 separate but similar problems:
- **missing 128 extra "setup(?)" samples**: I guess it's related how the xmaencoder inits MDCT but don't know enough to fix this. This is what actually breaks gapless as samples output +128 late so the very last subframe is broken.
- **missing skip samples**: I can patch this, but the above issue may be hard to see/test with this fixed though.

I did my best to explain in the bug report but may be hard to follow, could post more examples (also see the attached images).

I believe this also affects WMAPro and probably WMA lossless but I mainly tested XMA since I had the encoder.

**Comment by bxaimc** [comment]
Regarding the bad xma rips, this is mainly attributed to the lack of consistency checks Knurek has avoided when uploading to joshw. Back when I ran the tracker on hcs64, I was pretty anal about conversions of the sort and because of this I had rejected most of demon2warrior (Ultrafighter)’s rips since they were either bad rips or converted from one format to another. Once joshw opened up, Knurek grabbed just about any rip posted to the forum and the tracker couldn’t maintain oversight and thus had to shut down. The problem has gotten so bad over the years that it will take some time to rerip all of these games and I’m not sure anyone really wants to do that unfortunately.
